Applications/Classification/jni/main_func.cpp

```diff
@@ -0,0 +1,305 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ * @file	main_func.cpp
+ * @date	04 December 2019
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Jijoong Moon <jijoong.moon@samsung.com>
+ * @bug		No known bugs except for NYI items
+ * @brief	This is Classification Example with one FC Layer
+ *              The base model for feature extractor is mobilenet v2 with
+ * 1280*7*7 feature size. It read the Classification.ini in res directory and
+ * run according to the configureation.
+ *
+ */
+#include <climits>
+#include <cmath>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <queue>
+#include <sstream>
+#include <stdlib.h>
+#include <time.h>
+
+#include "layers.h"
+#include "neuralnet.h"
+#include "tensor.h"
+
+#define TRAINING true
+
+/**
+ * @brief     Data size for each category
+ */
+const unsigned int total_train_data_size = 100;
+
+const unsigned int total_val_data_size = 10;
+
+const unsigned int total_test_data_size = 100;
+
+const unsigned int buffer_size = 100;
+
+/**
+ * @brief     Number of category : Three
+ */
+const unsigned int total_label_size = 10;
+
+/**
+ * @brief     Max Epoch
+ */
+const unsigned int iteration = 3000;
+
+const unsigned int mini_batch = 32;
+
+const unsigned int feature_size = 62720;
+
+using namespace std;
+
+/**
+ * @brief     location of resources ( ../../res/ )
+ */
+string data_path;
+
+bool duplicate[total_label_size * total_train_data_size];
+bool valduplicate[total_label_size * total_val_data_size];
+
+/**
+ * @brief     step function
+ * @param[in] x value to be distinguished
+ * @retval 0.0 or 1.0
+ */
+float stepFunction(float x) {
+  if (x > 0.9) {
+    return 1.0;
+  }
+
+  if (x < 0.1) {
+    return 0.0;
+  }
+
+  return x;
+}
+
+/**
+ * @brief     Generate Random integer value between min to max
+ * @param[in] min : minimum value
+ * @param[in] max : maximum value
+ * @retval    min < random value < max
+ */
+static int rangeRandom(int min, int max) {
+  int n = max - min + 1;
+  int remainder = RAND_MAX % n;
+  int x;
+  do {
+    x = rand();
+  } while (x >= RAND_MAX - remainder);
+  return min + x % n;
+}
+
+/**
+ * @brief     load data at specific position of file
+ * @param[in] F  ifstream (input file)
+ * @param[out] outVec
+ * @param[out] outLabel
+ * @param[in] id th data to get
+ * @retval true/false false : end of data
+ */
+bool getData(std::ifstream &F, std::vector<float> &outVec,
```

Comment:
Q. Is there any rule how to arrange the order of arguments?

`param[in] param[out] param[out] param[in]` seems a bit unnatural to me

Reply:
http://www.doxygen.nl/manual/docblocks.html 
'For functions one can use the @param command to document the parameters and then use [in], [out], [in,out] to document the direction.'